src/cloudWatchLogs/activation.ts

```diff
@@ -39,11 +39,21 @@ export async function activate(context: vscode.ExtensionContext, configuration:
     context.subscriptions.push(
         vscode.workspace.onDidCloseTextDocument(doc => {
             if (doc.isClosed && doc.uri.scheme === CLOUDWATCH_LOGS_SCHEME) {
+                registry.disposeRegistryData(doc.uri)
                 registry.deregisterLog(doc.uri)
             }
         })
     )
 
+    context.subscriptions.push(
+        vscode.workspace.onDidChangeTextDocument((event: vscode.TextDocumentChangeEvent) => {
```

Comment:
nit: push() is variadic (takes unlimited args) so we can save a few lines by putting all of this disposables into one push() call.